pkg/controller/migcluster/validation.go

```diff
@@ -205,3 +226,37 @@ func (r ReconcileMigCluster) testConnection(cluster *migapi.MigCluster) error {
 
 	return nil
 }
+
+func (r *ReconcileMigCluster) saIsClusterAdmin(cluster *migapi.MigCluster) (bool, error) {
```

Comment:
This validation cannot be performed when any of the existing critical conditions exist.
```
if cluster.Status.HasCriticalCondition() {
	return nil
}
```

Also, let's be consistent with other validations.  The design pattern is to name the method: validate__() and to set the condition here (in the method) instead of returning a boolean.

Reply:
Fair enough. My thinking was that `saIsClusterAdmin` was just a helper function to `validateSaSecret`, but I can certainly make it its own top level validation and call it from `validate`.